pkg/stanza/operator/parser/keyvalue/keyvalue_test.go

```diff
@@ -488,15 +488,33 @@ key=value`,
 			false,
 		},
 		{
-			"invalid-pair",
+			"value-contains-delimiter",
 			func(kv *Config) {},
 			&entry.Entry{
 				Body: `test=text=abc`,
 			},
 			&entry.Entry{
+				Attributes: map[string]any{
+					"test": "text=abc",
+				},
 				Body: `test=text=abc`,
 			},
-			true,
+			false,
+			false,
+		},
+		{
+			"quoted-value-contains-delimiter",
+			func(kv *Config) {},
+			&entry.Entry{
+				Body: `msg="Message successfully sent at 2023-12-04 06:47:31.204222276 +0000 UTC m=+5115.932279346"`,
```

Comment:
Could we also have a test which involves 3+ key value pairs? This would validate we have the correct interaction with `pair_delimiter`, and help ensure we do not in the future limit the number of _pairs_ to 2.

Reply:
Added a `multiple-values-contain-delimiter` test case in 7b4ef32e60